snorkel/labeling/model/wrapper.py

```diff
@@ -0,0 +1,191 @@
+    device
+        What device to place the model on ('cpu' or 'cuda:0', for example)
+    metric
+        The metric to report with score()
+    tie_break_policy
+            Policy to break ties when converting probabilistic labels to predictions
+    n_epochs
+        The number of epochs to train (where each epoch is a single optimization step)
+    lr
+        Base learning rate (will also be affected by lr_scheduler choice and settings)
+    l2
+        Centered L2 regularization strength
+    optimizer
+        Which optimizer to use (one of ["sgd", "adam", "adamax"])
+    optimizer_config
+        Settings for the optimizer
+    lr_scheduler
+        Which lr_scheduler to use (one of ["constant", "linear", "exponential", "step"])
+    lr_scheduler_config
+        Settings for the LRScheduler
+    prec_init
+        LF precision initializations / priors
+    seed
+        A random seed to initialize the random number generator with
+    log_freq
+        Report loss every this many epochs (steps)
+    mu_eps
+        Restrict the learned conditional probabilities to [mu_eps, 1-mu_eps]
+    """
+
+    def __init__(
+        self,
+        L_train: np.ndarray,
+        cardinality: int = 2,
+        verbose: bool = True,
+        device: str = "cpu",
+        metric: str = "accuracy",
+        tie_break_policy: str = "abstain",
+        n_epochs: int = 100,
+        lr: float = 0.01,
+        l2: float = 0.0,
+        optimizer: str = "sgd",
+        optimizer_config: Optional[OptimizerConfig] = None,
+        lr_scheduler: str = "constant",
+        lr_scheduler_config: Optional[LRSchedulerConfig] = None,
+        prec_init: float = 0.7,
+        seed: int = np.random.randint(1e6),
+        log_freq: int = 10,
+        mu_eps: Optional[float] = None,
+        class_balance: Optional[List[float]] = None,
+        **kwargs: Any,
+    ) -> None:
+
+        self.L_train = L_train
+        self.cardinality = cardinality
+        self.verbose = verbose
+        self.device = device
+        self.metric = metric
+        self.tie_break_policy = tie_break_policy
+        self.n_epochs = n_epochs
+        self.lr = lr
+        self.l2 = l2
+        self.optimizer = optimizer
+        self.optimizer_config = (
+            OptimizerConfig()  # type: ignore
+            if optimizer_config is None
+            else optimizer_config
+        )
+        self.lr_scheduler = lr_scheduler
+        self.lr_scheduler_config = (
+            LRSchedulerConfig()  # type: ignore
+            if lr_scheduler_config is None
+            else lr_scheduler_config
+        )
+        self.prec_init = prec_init
+        self.seed = seed
+        self.log_freq = log_freq
+        self.mu_eps = mu_eps
+        self.class_balance = class_balance
+
+        self.label_model = LabelModel(
+            cardinality=self.cardinality, verbose=self.verbose, device=self.device
+        )
+
+    def fit(
+        self, L_dev: np.ndarray, Y_dev: Optional[np.ndarray] = None
```

Comment:
wait but L_dev isn't used at all...

Reply:
Yeah this is tied to the cv splits. Gridsearch expects X,Y to be the same for train and score, and does CV internally to figure out what to train on and what to score on. Might be fixed with predefined split